youtube_dl/extractor/tvp.py

```diff
@@ -250,3 +250,114 @@ def _real_extract(self, url):
         display_id, playlist_id = mobj.group('display_id', 'id')
         return self.playlist_result(
             self._entries(display_id, playlist_id), playlist_id)
+
+
+class TVPappIE(InfoExtractor):
+    IE_NAME = 'tvp:app'
+    # https://vod.tvp.pl/seriale,18/korona-krolow-jagiellonowie-odcinki,292227/odcinek-37,S01E37,392532
+    # https://vod.tvp.pl/programy,88/korona-krolow-taka-historia-odcinki,283653/odcinek-1,S02E01,386064
+    # https://vod.tvp.pl/filmy-dokumentalne,163/jurek,376503
+    # https://vod.tvp.pl/dla-dzieci,24/zubr-pompik-odcinki,282014/odcinek-1,S01E01,319853
+    # https://vod.tvp.pl/teatr-telewizji,202/barwy-uczuc,392420
+    # https://vod.tvp.pl/filmy-fabularne,136/rozlam,390638
+    _VALID_URL = r'https://vod.tvp.pl/(seriale|filmy-fabularne|programy|filmy-dokumentalne|dla-dzieci|teatr-telewizji|informacje-i-publicystyka),[0-9]+/([^,]+,[0-9]+/)?[^,]+,([^,]+,)?(?P<id>[0-9]+)'
+    _TESTS = [
+            {
+                # series
+                'url': 'https://vod.tvp.pl/seriale,18/korona-krolow-jagiellonowie-odcinki,292227/odcinek-37,S01E37,392532',
+                'info_dict': {
+                    'id': '392532',
+                    'ext': 'mp4',
+
+                    'series_id': 292227,
+                    'title': 'Korona królów. Jagiellonowie - Episode 37 - odc. 37 – Branka',
+                    'description': 'Wiosna 1407. Anna z Goraja znika z Wawelu. Jej matka - Beata z Bożego Daru rozpacza. Anna Cylejska  zarządza poszukiwania, a Sofia tajemniczo się uśmiecha. Elżbieta i Katarzyna Gorajskie wyjawiają, że ich siostra pojechała do klasztoru klarysek w Sączu. Na Wawel dociera wieść o śmierci Wielkiego Mistrza Konrada von Jungingena. Rozpoczyna się walka o władzę w Malborku. Jan Falkenberg ma list do papieża, w którym wyjawia sekret Trąby. Bdzigost i Ciołek zdobywają to pismo. Król Jagiełło poznaje wreszcie sekret Mikołaja Trąby i znajduje sposób, by go ochronić.',
+                    'thumbnail': r're:^https?://.*\.jpg$',
+                    'episode_number': 37,
+                },
+                'params': {
+                    'skip_download': True,
+                    'format': 'dash-f1-v1-x3',
```

Comment:
These options are contradictory, but that might be OK if the point is to skip downloads in CI tests where the source IP might be geo-blocked, but to remove the `skip_download` option for local testing.

Generally, skip the download, or at least don't check a file MD5, if the media download is adaptive, as here. It's probably best to specify a generic format rather than a specific one, as here. The download test harness uses `'format': 'best'` but, if the site has no combined formats, use (eg) `'format': 'bestvideo'`.